src/status_im/ui/screens/wallet/send/views/amount.cljs

```diff
@@ -0,0 +1,314 @@
+                      token-symbol)))
+
+(defn create-initial-state [{:keys [symbol decimals]} amount]
+  {:input-amount  (when amount
+                    (when-let [amount' (money/internal->formatted amount symbol decimals)]
+                      (str amount')))
+   :inverted      false
+   :edit-gas      false
+   :error-message nil})
+
+(defn input-currency-symbol [{:keys [inverted] :as state} {:keys [symbol] :as coin} {:keys [code]}]
+  {:pre [(boolean? inverted) (keyword? symbol) (string? code)]}
+  (if-not (:inverted state) (wallet.utils/display-symbol coin) code))
+
+(defn converted-currency-symbol [{:keys [inverted] :as state} {:keys [symbol] :as coin} {:keys [code]}]
+  {:pre [(boolean? inverted) (keyword? symbol) (string? code)]}
+  (if (:inverted state) (wallet.utils/display-symbol coin) code))
+
+(defn valid-input-amount? [input-amount]
+  (and (not (string/blank? input-amount))
+       (not (:error (wallet.db/parse-amount input-amount 100)))))
+
+(defn converted-currency-amount [{:keys [input-amount inverted]} token fiat-currency prices]
+  (when (valid-input-amount? input-amount)
+    (if-not inverted
+      (some-> (common/token->fiat-conversion prices token fiat-currency input-amount)
+              (money/with-precision 2))
+      (some-> (common/fiat->token-conversion prices token fiat-currency input-amount)
+              (money/with-precision 8)))))
+
+(defn converted-currency-phrase [state token fiat-currency prices]
+  (str (if-let [amount-bn (converted-currency-amount state token fiat-currency prices)]
+         (str amount-bn)
+         "0")
+       " " (converted-currency-symbol state token fiat-currency)))
+
+(defn current-token-input-amount [{:keys [input-amount inverted] :as state} token fiat-currency prices]
+  {:pre [(map? state) (map? token) (map? fiat-currency) (map? prices)]}
+  (when input-amount
+    (when-let [amount-bn (if inverted
+                           (common/fiat->token-conversion prices token fiat-currency input-amount)
+                           (money/bignumber input-amount))]
+      amount-bn
+      (money/formatted->internal amount-bn (:symbol token) (:decimals token)))))
+
+(defn update-input-errors [{:keys [input-amount inverted] :as state} token fiat-currency prices]
+  {:pre [(map? state) (map? token) (map? fiat-currency) (map? prices)]}
+  (let [{:keys [_value error]}
+        (wallet.db/parse-amount input-amount
+                                (if inverted 2 (:decimals token)))]
+    (if-let [error-msg
+             (cond
+               error error
+               (not (money/sufficient-funds? (current-token-input-amount state token fiat-currency prices)
+                                             (:amount token)))
+               (i18n/label :t/wallet-insufficient-funds)
+               :else nil)]
+      (assoc state :error-message error-msg)
+      state)))
+
+(defn update-input-amount [state input-str token fiat-currency prices]
```

Comment:
can we somehow split this function? a bit hard to read

Reply:
sure